pkg/controller/operators/catalog/step_ensurer.go

```diff
@@ -109,6 +109,29 @@ func (o *StepEnsurer) EnsureSecret(operatorNamespace, planNamespace, name string
 	return
 }
 
+// EnsureBundleSecret creates user-specified secrets from the bundle. Called when StepResource.Secret is true
+func (o *StepEnsurer) EnsureBundleSecret(namespace string, secret *corev1.Secret) (status v1alpha1.StepStatus, err error) {
+	_, createErr := o.kubeClient.KubernetesInterface().CoreV1().Secrets(namespace).Create(secret)
+	if createErr == nil {
+		status = v1alpha1.StepStatusCreated
+		return
+	}
+
+	if !k8serrors.IsAlreadyExists(createErr) {
+		err = errorwrap.Wrapf(createErr, "error updating secret: %s", secret.GetName())
+		return
+	}
+
+	secret.SetNamespace(namespace)
+	if _, updateErr := o.kubeClient.UpdateSecret(secret); updateErr != nil {
```

Comment:
This will cause any annotations/changes applied to the secret to be lost - unsure if that matter but it's worth calling out.

Reply:
Also - does this work without the `ResourceID`?

Reply:
That's a good point -- this is existing code that hasn't been changed in this PR though. I can make a note however.

Reply:
Could we make a unit test instead of a note?